src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -442,9 +441,19 @@ def resize_subvolume(self, fs_handle, **kwargs):
                     raise VolumeException(
                         -errno.ENOENT, "Subvolume '{0}' not found, create it with " \
                         "'ceph fs subvolume create' before resizing subvolumes".format(subvolname))
+
+                try:
+                    newsize = int(newsize)
+                except ValueError:
+                    newsize = newsize.lower()
+                    nsize, usedbytes = sv.resize_infinite(subvolpath, newsize)
```

Comment:
I think `try except else` will be easier to read. https://stackoverflow.com/questions/16138232/is-it-a-good-practice-to-use-try-except-else-in-python

```
try:
      newsize = int(newsize)
except ValueError:
      nsize, usedbytes = sv.resize_infinite(subvolpath, newsize.lower())
      ret= 0, json.dumps([{'bytes_used': usedbytes}, {'bytes_quota': nsize}, {'bytes_pcent': "undefined"}], indent=2), ""
else:
      nsize, usedbytes = sv.resize_subvolume(subvolpath, newsize, kwargs['no_shrink'])
      ret = 0, json.dumps([{'bytes_used': usedbytes}, {'bytes_quota': nsize},
                                     {'bytes_pcent': '{0:.2f}'.format((float(usedbytes) / nsize) * 100.0)}], indent=2), ""
```

Reply:
oh yes, that's right.